### PR TITLE
102 create edit profile routes

### PR DESCRIPTION
**Pull request by alaaahmade**
102 patch create edit profile routes #102

### CODE REVIEW

**Review by Saeed99Madi — APPROVED** [review]
Great Work Alaa @alaaahmade

File: server/services/stadiums.ts [hunk 1]
```diff
@@ -109,3 +110,145 @@ export const getStadiumProfileService = async (
...
+    return {
+      status: 401,
+      data: 'UnAuthorize',
+    };
+  }
+
+  const check = await Gallery.findAll({ where: { id, StadiumId } });
+  if (!check.length) {
+    return {
+      status: 401,
```
Comment (ahmedirheem) [inline server/services/stadiums.ts:174]: 404 is the status of not found

File: server/services/stadiums.ts [hunk 2]
```diff
@@ -109,3 +110,145 @@ export const getStadiumProfileService = async (
...
+  }
+
+  const galleries = await Gallery.update(
+    { image },
+    { where: { id, StadiumId }, returning: true },
+  );
+
+  return {
+    status: 200,
+    data: galleries,
```
Comment (ahmedirheem) [inline server/services/stadiums.ts:186]: galleries consist of two things, the number of updated rows and the updated values itself, and here we just need updated values itself

File: server/services/stadiums.ts [hunk 3]
```diff
@@ -109,3 +110,145 @@ export const getStadiumProfileService = async (
...
+      data: 'UnAuthorize',
+    };
+  }
+
+  const check = await Gallery.findOne({
+    where: { StadiumId: +StadiumId, id: +ImageId },
+  });
+  if (!check) {
+    return {
+      status: 401,
```
Comment (ahmedirheem) [inline server/services/stadiums.ts:244]: 404 is the status of not found

File: server/services/stadiums.ts [hunk 4]
```diff
@@ -109,3 +110,145 @@ export const getStadiumProfileService = async (
...
+      data: 'هذه الصورة غير موجودة',
+    };
+  }
+
+  const result = await Gallery.destroy({
+    where: { StadiumId: +StadiumId, id: +ImageId },
+  });
+
+  return { status: 204, data: result };
```
Comment (ahmedirheem) [inline server/services/stadiums.ts:253]: `result` here is the number of deleted rows, what the benefit from it to send it in the response

**Review by ahmedirheem — CHANGES_REQUESTED** [review]

**Review by ahmedirheem — APPROVED** [review]
Good Job Alaa

File: server/controllers/stadiums.ts [hunk 5]
```diff
@@ -1,10 +1,16 @@
 import {
+  UpdateStadiumDataService,
   getAllStadiumsService,
   getStadiumDetailsService,
   getStadiumMatchesService,
   getStadiumProfileService,
 } from '../services';
 import { RequestHandler, Request, Response } from 'express';
```
Comment (Emanalabsi) [inline server/controllers/stadiums.ts:8]: move this to the top

File: server/controllers/stadiums.ts [hunk 6]
```diff
@@ -1,10 +1,16 @@
...
+import {
+  AddStadiumImageService,
+  UpdateStadiumGalleryService,
+  deleteStadiumImageService,
+} from '../services/stadiums';
 
 export const getAllStadiums: RequestHandler = async (
   req: Request,
```
Comment (Emanalabsi) [inline server/controllers/stadiums.ts:16]: where is res: response! how it worked?

File: server/controllers/stadiums.ts [hunk 7]
```diff
@@ -40,3 +46,39 @@ export const getStadiumProfile: RequestHandler = async (
...
+  res.status(response.status).json(response);
+};
+
+export const UpdateStadiumGallery: RequestHandler = async (
+  req: Request,
+  res: Response,
+): Promise<void> => {
+  const response = await UpdateStadiumGalleryService(req);
+
+  res.status(response.status).json(response);
```
Comment (Emanalabsi) [inline server/controllers/stadiums.ts:65]: response?.status, there might be no response status

**Review by Emanalabsi — CHANGES_REQUESTED** [review]

**Review by Emanalabsi — APPROVED** [review]

**Pull request merged**